chain/client/src/client_actor.rs

```diff
@@ -1782,11 +1782,17 @@ impl ClientActor {
         let is_syncing = self.client.sync_status.is_syncing();
         let head = unwrap_or_return!(self.client.chain.head());
         let validator_info = if !is_syncing {
-            let validators = unwrap_or_return!(self
+            let block_and_chunk_validators = unwrap_or_return!(self
                 .client
                 .runtime_adapter
                 .get_epoch_block_producers_ordered(&head.epoch_id, &head.last_block_hash));
-            let num_validators = validators.len();
+            let chunk_validators = unwrap_or_return!(self
+                .client
+                .runtime_adapter
+                .get_epoch_chunk_producers(&head.epoch_id));
+            let num_validators = chunk_validators.len();
+            let num_chunk_only_validators = num_validators - block_and_chunk_validators.len();
```

Comment:
Looks like the logic behind those calculations doesn’t quite work.  This is panicking with an overflow.

Reply:
I don’t think that’s a good solution.  After all, it’s not possible to have negative number of chunk only validators.  At the very least this should be `num_validators.saturating_sub(block_and_chunk_validators.len()`, but I’d rather we understood why this can give negative result.

Reply:
`num_validators.saturating_sub(block_and_chunk_validators.len())` results in an error:
```
mismatched types [E0308] expected `isize`, found `usize`
```

Reply:
Think I fixed it using [checked_sub](https://doc.rust-lang.org/std/primitive.usize.html#method.checked_sub) in [61e0d38](https://github.com/near/nearcore/pull/7788/commits/61e0d38642e14d8abeaafa31b737d083b51fa8a7)

Reply:
To use saturating_sub you need to use usize for the number.

I still don't think this is correct.  We're just subtracting numbers which we don’t understand it sounds to me.  How is it possible that the number is negative?

Reply:
I don't think the number ever is negative in a live scenario (I've never seen it), but maybe the compiler fails bc it *_could_* be negative. Gave the `saturating_sub` another try and it's working. Not sure why it was erroring for me earlier

Reply:
No, it’s not a compilation issue. The tests were filing so the number did get negative in tests.